R/splatPopParams-methods.R

```diff
@@ -0,0 +1,166 @@
+    params <- new("splatPopParams")
+    params <- setParams(params, ...)
+    
+    checkmate::assertClass(params, "splatPopParams")
+
+    for (pkg in c("VariantAnnotation", "preprocessCore")) {
+        if (!requireNamespace(pkg, quietly = TRUE)) {
+            stop("The splatPop simulation requires the ", pkg, " package.")
+        }
+    }
+
+    return(params)
+}
+
+
+#' @importFrom checkmate checkInt checkIntegerish checkNumber checkNumeric
+#' checkFlag
+setValidity("splatPopParams", function(object) {
+
+    v <- getParams(object, c(slotNames(object)))
+
+    checks <- c(eqtl.n = checkNumber(v$eqtl.n, lower = 0),
+                eqtl.dist = checkInt(v$eqtl.dist, lower = 1),
+                eqtl.maf.min = checkNumber(v$eqtl.maf.min, lower = 0,
+                                           upper = 0.5),
+                eqtl.maf.max = checkNumber(v$eqtl.maf.max, lower = 0,
+                                           upper = 0.5),
+                eqtl.ES.shape = checkNumber(v$eqtl.ES.shape, lower = 0),
+                eqtl.ES.rate = checkNumber(v$eqtl.ES.rate, lower = 0),
+                eqtl.group.specific = checkNumber(v$eqtl.group.specific,
+                                                 lower = 0, upper = 1),
+                pop.mean.shape = checkNumber(v$pop.mean.shape, lower = 0),
+                pop.mean.rate = checkNumber(v$pop.mean.rate, lower = 0),
+                pop.cv.bins = checkInt(v$pop.cv.bins, lower = 1),
+                pop.cv.param = checkDataFrame(v$pop.cv.param),
+                similarity.scale = checkNumber(v$similarity.scale, lower = 0))
+
+    if (all(checks == TRUE)) {
+        valid <- TRUE
+    } else {
+        valid <- checks[checks != TRUE]
+        valid <- paste(names(valid), valid, sep = ": ")
+    }
+
+    return(valid)
+})
+
+
+#' @importFrom methods callNextMethod
+setMethod("show", "splatPopParams", function(object) {
+
+    pp <- list("Population params:" = c("(mean.shape)" = "pop.mean.shape",
+                                        "(mean.rate)" = "pop.mean.rate",
+                                        #"[group.prop]" = "group.prop",
+                                        "[similarity.scale]" = "similarity.scale",
+                                        "[cv.bins]" = "pop.cv.bins",
+                                        "(cv.params)" = "pop.cv.param"),
+               "eQTL params:" = c("[eqtl.n]"    = "eqtl.n",
+                                  "[eqtl.dist]" = "eqtl.dist",
+                                  "[eqtl.maf.min]" = "eqtl.maf.min",
+                                  "[eqtl.maf.max]" = "eqtl.maf.max",
+                                  "[eqtl.group.specific]" = "eqtl.group.specific",
+                                  "(eqtl.ES.shape)" = "eqtl.ES.shape",
+                                  "(eqtl.ES.rate)" = "eqtl.ES.rate"))
+
+    callNextMethod()
+    showPP(object, pp)
+})
+
+
+#' @rdname setParam
+setMethod("setParam", "splatPopParams", function(object, name, value) {
```

Comment:
I think most of this should be covered by the `SplatParams` method, but need to check.